main.tf

```diff
@@ -106,7 +106,7 @@ resource "aws_instance" "default" {
 }
 
 resource "aws_eip" "default" {
-  count    = "${signum(length(var.associate_public_ip_address)) == 1 ? 1 : 0}"
+  count    = "${var.associate_public_ip_address ? 1 : 0}"
   instance = "${aws_instance.default.id}"
   vpc      = true
 }
```

Comment:
Update `ansible` `envs` with corresponding IP address selection.
See https://github.com/cloudposse/tf_instance/pull/6/files#diff-7a370d8342e7203b805911c92454f0f4R118

Reply:
```
module "ansible" {
   source    = "git::https://github.com/cloudposse/tf_ansible.git?ref=tags/0.3.4"
   arguments = "${var.ansible_arguments}"
   envs      = "${compact(concat(var.ansible_envs, list("host=${var.associate_public_ip_address ? aws_instance.default.public_dns : aws_instance.default.private_dns }")))}"
   playbook  = "${var.ansible_playbook}"
   dry_run   = "${var.ansible_dry_run}"
 }
```

Reply:
Currently `ansible` module uses improper `host` env.
It should be updated accordingly to EIP feature.